### PR TITLE
Hide cursor

### PR DESCRIPTION
**Pull request by Tom94**
Updates the framework and manually hides the OS cursor.

Also contains other minor fixes, and makes the direction of the cursor nicer while dragging.

### CODE REVIEW

File: osu.Game/Graphics/Cursor/MenuCursor.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ protected override bool OnMouseUp(InputState state, MouseUpEventArgs args)
             if (!state.Mouse.HasMainButtonPressed)
             {
                 ((Cursor)ActiveCursor).AdditiveLayer.FadeOut(500, EasingTypes.OutQuint);
-                ActiveCursor.RotateTo(0, 600, EasingTypes.OutElasticHalf);
+                ActiveCursor.RotateTo(0, 600 * Math.Abs(ActiveCursor.Rotation / 720 + 1), EasingTypes.OutElasticHalf);
```
Comment (Tom94) [inline osu.Game/Graphics/Cursor/MenuCursor.cs]: +1 outside of the `Math.Abs`

File: osu.Desktop.VisualTests/VisualTestGame.cs [hunk 2]
```diff
@@ -19,5 +20,11 @@ protected override void LoadComplete()
             // we depend on some dependencies to be loaded within OsuGameBase.load().
             Add(new TestBrowser());
         }
+
+        public override void SetHost(GameHost host)
```
Comment (Tom94) [inline osu.Desktop.VisualTests/VisualTestGame.cs]: Why not just override `OsuGameBase.SetHost`?

Comment (peppy) [inline osu.Desktop.VisualTests/VisualTestGame.cs]: headless

Comment (Tom94) [inline osu.Desktop.VisualTests/VisualTestGame.cs]: Can check for `Host.Window == null`?

Comment (Tom94) [inline osu.Desktop.VisualTests/VisualTestGame.cs]: IIRC some code in `OsuGameBase` already does that anyway.

Comment (peppy) [inline osu.Desktop.VisualTests/VisualTestGame.cs]: would rather not :/

**Pull request merged**